applications/hotornot/src/hon_util.erl

```diff
@@ -119,7 +119,7 @@ sort_rate(RateA, RateB) ->
 
     case PrefixA =:= PrefixB of
         'true' ->
-            wh_json:get_integer_value(<<"weight">>, RateA, 100) >
+            wh_json:get_integer_value(<<"weight">>, RateA, 100) <
                 wh_json:get_integer_value(<<"weight">>, RateB, 100);
         'false' ->
             PrefixA > PrefixB
```

Comment:
The reason I was looking at this code is to allow rates that only reply to specific resellers. This is currently in testing.
The code I actually have is:

````
%% Return true if RateA is more applicable than RateB
-spec sort_rate(wh_json:object(), wh_json:object()) -> boolean().
sort_rate(RateA, RateB) ->
    PrefixA = byte_size(wh_json:get_binary_value(<<"prefix">>, RateA)),
    PrefixB = byte_size(wh_json:get_binary_value(<<"prefix">>, RateB)),

    %% Do these rates only apply to a reseller?
    ResellerA = wh_json:get_value(?EXCLUDE_TO_RESELLERS_KEY, RateA) =/= 'undefined',
    ResellerB = wh_json:get_value(?EXCLUDE_TO_RESELLERS_KEY, RateB) =/= 'undefined',

    case {{ResellerA, ResellerB}, PrefixA =:= PrefixB} of
        %% Prefer a rate specific to the reseller
        {{'true', 'false'}, _} ->
            'true';
        {{'false', 'true'}, _} ->
            'false';
        {_, 'true'} ->
            %% If prefix length is equal then prefer rates with lower weights
            wh_json:get_integer_value(<<"weight">>, RateA, 100) <
                wh_json:get_integer_value(<<"weight">>, RateB, 100);
        {_, 'false'} ->
            %% Prefer longer prefixes
            PrefixA > PrefixB
    end.
````

Rates that don't apply to this account will have already be filtered out by this time.
So, what I have will always prefer a rate that is exclusive to the reseller, if both or neither are exclusive:

A prefix length, B prefix length | weight A, weight B | result
------------ | ------------- | -------------
A < B  | * | B
A > B  | * | A
A = B  | A < B | A
A = B  | A > B | B
A = B  | A = B | A?